Update rebalance zh/en docs (#101)
apache/incubator-pegasus-website: _docs/en/administration/rebalance.md

permalink: administration/rebalance
---

This document mainly introduces the concepts, usage, and design of rebalance in Pegasus.

## Concept Section
In Pegasus, rebalance mainly includes the following aspects:
1. If a certain partition replica does not meet the requirement of one primary and two secondaries, a node needs to be selected to complete the missing replica. This process in Pegasus is called `cure`.

> 2. After all partitions meet the requirement of one primary and two secondaries, the number of replicas on each replica server in the cluster is adjusted to try to keep the number of replicas served by each machine at a similar level. This process in Pegasus is called `balance`.

Reviewer: ```suggestion
2. After all partitions have 3 replicas each, all replicas need to be distributed evenly across the replica servers. This process in Pegasus is called `balance`.
```

> 3. If a replica server has multiple disks mounted and provided to Pegasus for use through the configuration file `data_dirs`, the replica server should try to keep the number of replicas on each disk at a similar level.

Reviewer: ```suggestion
3. If a replica server has multiple disks mounted and provided to Pegasus through the configuration file `data_dirs`, the replica server should try to keep the number of replicas on each disk at a similar level.
```

   
Based on these points, Pegasus has introduced some concepts to conveniently describe these situations:
1. The Health Status of Partition
   Pegasus has defined several health statuses for partitions:  
   * 【fully healthy】: Healthy, fully meeting the requirement of one primary and two secondaries.
   * 【unreadable】: The partition is unreadable. It means that the partition lacks a primary, but there is one or two secondaries.
   * 【readable but unwritable】: The partition is readable but not writable. It means that only one primary remains, and both secondary replicas are lost.
   * 【readable and writable but unhealthy】: The partition is both readable and writable, but still not healthy. It means that one secondary is missing among the three replicas.
   * 【dead】: All replicas of the partition are unavailable, also known as the DDD state.
![pegasus-healthy-status](/assets/images/pegasus-healthy-status.png){:class="img-responsive"}

>    When checking the status of the cluster, tables, and partitions through the Pegasus shell, you will often see the overall statistics or individual descriptions of the health conditions of the partitions. For example, by using the ls -d command, you can see the number of partitions in different health conditions for each table, including the following:

Reviewer: ```suggestion
   When checking the status of the cluster, tables, and partitions through the Pegasus shell, you will often see the overall statistics or individual descriptions of the health conditions of the partitions. For example, by using the `ls -d` command, you can see the number of partitions in different health conditions for each table, including the following:
```

Author: Thanks, all suggestions committed.
